release/ray_release/config.py

```diff
@@ -47,10 +58,29 @@ def read_and_validate_release_test_collection(
 ) -> List[Test]:
     """Read and validate test collection from config file"""
     with open(config_file, "rt") as fp:
-        test_config = yaml.safe_load(fp)
+        tests = parse_test_definition(yaml.safe_load(fp))
+
+    validate_release_test_collection(tests, schema_file=schema_file)
+    return tests
 
-    validate_release_test_collection(test_config, schema_file=schema_file)
-    return test_config
+
+def parse_test_definition(test_definitions: List[TestDefinition]) -> List[Test]:
+    tests = []
+    for test_definition in test_definitions:
+        if "variations" not in test_definition:
+            tests.append(test_definition)
+            continue
+        variations = test_definition.pop("variations")
+        if not variations:
+            raise ReleaseTestConfigError(
+                f'Empty variations defined in test {test_definition["name"]}',
+            )
+        for variation in variations:
+            test = copy.deepcopy(test_definition)
+            test["name"] = f'{test["name"]}.{variation.pop("__suffix__")}'
```

Comment:
report an error when `__suffix__` is missing?

or.. maybe just emit variation without suffix?

like 

````
variations:
- name: example
  variations:
  - ~ # emits "example"
  - __suffix__: gce # emits "example.gce"

Reply:
Sound good. I like throwing error better, in case folks truly forget putting a __suffix__ ;)